athenet/network.py

```diff
@@ -2,45 +2,69 @@
 
 import timeit
 import numpy as np
+from enum import Enum
 
 import theano
 import theano.tensor as T
 
 from athenet.layers import WeightedLayer, ConvolutionalLayer
+from athenet.utils import overwrite, save_data_to_pickle
+
+
+class TrainConfig(object):
+    """Class storing information about training process."""
+    def __init__(self):
+        self.n_epochs = None
+        self.batch_size = None
+
+        self.learning_rate = None
+        self.momentum = 0.
+        self.weight_decay = 0.
+
+        self.val_interval = 1
+        self.val_units = 'epochs'
+
+        self.val_range = 4
+        self.learning_rate_decay = 1.
+        self.learning_rate_threshold = 1.
+
+    def __str__(self):
+        output = '{} epochs, minibatch size = {}, learning rate = {}'\
+                 .format(self.n_epochs, self.batch_size, self.learning_rate)
+        if self.momentum:
+            output += ', momentum = {}'.format(self.momentum)
+        if self.weight_decay:
+            output += ', weight_decay = {}'.format(self.weight_decay)
+        if self.learning_rate_decay != 1.:
+            output += '\nLearning rate decay = {}: threshold = {}, '\
+                      'validation range = {}'\
+                      .format(self.learning_rate_decay,
+                              self.learning_rate_threshold,
+                              self.val_range)
+        return output
 
 
 class Network(object):
     """Neural network."""
-
-    verbosity = 0
-
-    # Early stopping parameters
-    initial_patience = 10000
-    patience_increase = 2
-
     def __init__(self, layers):
         """Create neural network.
 
         layers: List of network's layers.
         """
-        self._data_accuracy = None
-        self._test_data_accuracy = None
-        self._val_data_accuracy = None
         self._batch_size = None
         self._data_loader = None
-
-        self.params = None
-        self.output = None
-        self.answer = None
-        self.train_output = None
+        self.answers = None
         self.get_output = None
+        self._accuracy = None
+        self._accuracy_config = None
 
+        self.snapshot_name = 'network'
+        self.snapshot_interval = 0
+        self.verbosity = 1
```

Comment:
Put a comment on what can verbosity be. (1-3?). I would consider creating a function for checking verbosity level for internal use but it's up to you.